dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -512,21 +515,31 @@ protected boolean invokeTraceRouters(RoutedMessage routedTrace) {
 
         boolean retMe = true;
         LogRecord logRecord = routedTrace.getLogRecord();
-        if (logRecord != null) {
-            Level level = logRecord.getLevel();
-            int levelValue = level.intValue();
-            if (levelValue < Level.INFO.intValue()) {
-                String levelName = level.getName();
-                if (!(levelName.equals("SystemOut") || levelName.equals("SystemErr"))) { //SystemOut/Err=700
-                    WsTraceRouter internalTrRouter = internalTraceRouter.get();
-                    if (internalTrRouter != null) {
-                        retMe &= internalTrRouter.route(routedTrace);
-                    } else {
-                        earlierTraces.add(routedTrace);
+
+        /*
+         * Avoid any feedback traces that are emitted after this point.
+         * The first time the counter increments is the first pass-through.
+         * The second time the counter increments is the second pass-through due
+         * to trace emitted. We do not want any more pass-throughs.
+         */
+        if (!(counter.incrementCount() > 2)) {
+            if (logRecord != null) {
+                Level level = logRecord.getLevel();
+                int levelValue = level.intValue();
+                if (levelValue < Level.INFO.intValue()) {
+                    String levelName = level.getName();
+                    if (!(levelName.equals("SystemOut") || levelName.equals("SystemErr"))) { //SystemOut/Err=700
+                        WsTraceRouter internalTrRouter = internalTraceRouter.get();
+                        if (internalTrRouter != null) {
+                            retMe &= internalTrRouter.route(routedTrace);
+                        } else {
+                            earlierTraces.add(routedTrace);
+                        }
                     }
                 }
             }
         }
+        counter.decrementCount();
```

Comment:
counter - should decrement counter in finally block to avoid possibility that an exception terminates method without decrementing